tests/financial/test_national_stock_exchange_of_india.py

```diff
@@ -103,10 +135,17 @@ def test_maha_shivratri(self):
             "2024-03-08",
             "2025-02-26",
         )
-        self.assertNonObservedHolidayName(name, range(2001, 2036))
+        self.assertNonObservedHolidayName(name, range(2007, 2036))
+        self.assertNoNonObservedHolidayName(name, range(2001, 2007))
         self.assertNoHoliday(
+            "2013-03-10",
             "2023-02-18",
         )
+        self.assertNoHolidayName(
+            name,
+            2013,
+            2023,
+        )
```

Comment:
```suggestion
        self.assertNoHolidayName(name, 2013, 2023)
```
If we add such check, then it needed for all holidays.

Reply:
okay, if its not required i can avoid it.